configure.ac: Use PKG_CHECK_MODULES to detect libxml2

hughmcmaster opened the pull request:
Debian has begun removing `xml2-config` from its libxml2-dev package, meaning `configure` fails to find libxml2.

This patch modifies `configure.ac` to use `PKG_CHECK_MODULES` to detect libxml2.

rfm commented:
Unless I'm missing something, this patch replaces the existing mechanism and would therefore break libxml2 detection on all older platforms (and possibly some modern ones).

I'd very much like to see a patch which tries pkgconfig first and then falls back to use the existing detection mechanism if pkgconfig is not available or can't find the library.

hughmcmaster commented:
> 
> 
> Unless I'm missing something, this patch replaces the existing mechanism and would therefore break libxml2 detection on all older platforms (and possibly some modern ones).

libs-base already attempts to use pkg-config, and ICU is detected solely by pkg-config. Modernising would be ideal. However...

> I'd very much like to see a patch which tries pkgconfig first and then falls back to use the existing detection mechanism if pkgconfig is not available or can't find the library.

Okay, I've updated the patch with this in mind.

Note that I've removed the save and restore CFLAGS/LIBS lines, as I could not find any evidence that AM_PATH_XML clobbers $CFLAGS or $LIBS. Besides, the existing restore code was useless, since configure terminated anyway. And, if $LIBS was clobbered, `LIBS="$XML_LIBS $LIBS` would be broken too.

hughmcmaster commented:
Hi @rfm, will you have time to look at my patch this week?

rfm commented:
> On 4 Feb 2020, at 10:22, Hugh McMaster <notifications@github.com> wrote:
> 
> Hi @rfm, will you have time to look at my patch this week?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

I'm away on a business trip, but should be able to look at it on friday.
Or, Fred might give it a loom and merge it earlier if he has time.
